README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# MEDITS

<!-- badges: start -->

<!-- badges: end -->

The MEDITS packages was realized to work with survey data in the format
of the MEDITS project. The package uses TA-, TB-, and TC-like tables.
Functions using TE and TL table are not already implemented. TA table
contains hauls’ information; TB contains catch data; TC contains
aggregated biological data. Examples of TA, TB and TC table were added
to the package in order to execute the example codes.

## Installation

You can install the released version of MEDITS from
[CRAN](https://CRAN.R-project.org) with:

``` r
install.packages("MEDITS")
```

## Working with coordinates (TA file)

The MEDITS format uses a specific format for the geographic coordinates.
Two functions were introduced to convert the coordinates respectively
from the MEDITs format to decimal degrees format and viceversa.

``` r
# Coordinates conversion from MEDITS format to decimal degrees
library(MEDITS)
dd <- MEDITS.to.dd(TA)
head(dd)
#>   TYPE_OF_FILE COUNTRY AREA VESSEL  GEAR RIGGING DOORS YEAR MONTH DAY
#> 1           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#> 2           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#> 3           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#> 4           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#> 5           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#> 6           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#>   HAUL_NUMBER CODEND_CLOSING PART_OF_THE_CODEND SHOOTING_TIME
#> 1           1              S                  S          1114
#> 2           2              S                  S           948
#> 3           3              S                  S          1500
#> 4           4              S                  S          1355
#> 5           5              S                  S           502
#> 6           6              S                  S           615
#>   SHOOTING_QUADRANT SHOOTING_LATITUDE SHOOTING_LONGITUDE SHOOTING_DEPTH
#> 1                 1          40.96233           13.66683            376
#> 2                 1          41.03167           13.69167            109
#> 3                 1          40.90900           13.82950            113
#> 4                 1          40.95750           13.78783             95
#> 5                 1          40.81617           13.96200             77
#> 6                 1          40.91733           13.93683             25
#>   HAULING_TIME HAULING_QUADRANT HAULING_LATITUDE HAULING_LONGITUDE
#> 1         1214                1         40.92067          13.70017
#> 2         1018                1         41.01117          13.70483
#> 3         1530                1         40.88800          13.85900
#> 4         1425                1         40.94050          13.80650
#> 5          532                1         40.83967          13.95783
#> 6          645                1         40.90317          13.91733
#>   HAULING_DEPTH HAUL_DURATION VALIDITY COURSE RECORDED_SPECIES DISTANCE
#> 1           331            60        V      R                3     5500
#> 2           112            30        V      R                3     2593
#> 3           119            30        V      R                3     2926
#> 4            97            30        V      R                3     2500
#> 5            75            30        V      R                3     2759
#> 6            23            30        V      R                3     2630
#>   VERTICAL_OPENING WING_OPENING GEOMETRICAL_PRECISION BRIDLES_LENGTH
#> 1               19          191                     E            150
#> 2               19          182                     E            100
#> 3               19          182                     E            100
#> 4               19          182                     E            100
#> 5               19          182                     E            100
#> 6               19          163                     E            100
#>   WARP_LENGTH WARP_DIAMETER HYDROLOGICAL_STATION OBSERVATIONS
#> 1        1300            14                    0            0
#> 2         550            14                    0            0
#> 3         550            14                    0            0
#> 4         550            14                    0            0
#> 5         450            14                    0            0
#> 6         300            14                    0            0
#>   BOTTOM_TEMPERATURE_BEGINNING BOTTOM_TEMPERATURE_END MEASURING_SYSTEM
#> 1                         14.0                   14.0               VA
#> 2                         14.3                   14.2               VA
#> 3                         14.3                   14.0               VA
#> 4                         14.6                   14.3               VA
#> 5                         14.9                   14.9               VA
#> 6                         18.1                   18.6               VA
#>   NUMBER_OF_THE_STRATUM BOTTOM_SALINITY_BEGINNING BOTTOM_SALINITY_END
#> 1                 13404                        NA                  NA
#> 2                 13403                        NA                  NA
#> 3                 13403                        NA                  NA
#> 4                 13402                        NA                  NA
#> 5                 13402                        NA                  NA
#> 6                 13401                        NA                  NA
#>   MEASURING_SYSTEM_SALINITY
#> 1                        NA
#> 2                        NA
#> 3                        NA
#> 4                        NA
#> 5                        NA
#> 6                        NA

# Coordinates conversion from decimal degrees to MEDITS format
med <- dd.to.MEDITS(dd)
head(med)
#>   TYPE_OF_FILE COUNTRY AREA VESSEL  GEAR RIGGING DOORS YEAR MONTH DAY
#> 1           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#> 2           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#> 3           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#> 4           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#> 5           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#> 6           TA     ITA   10    PEC GOC73    GC73  WHS8 2007     8  11
#>   HAUL_NUMBER CODEND_CLOSING PART_OF_THE_CODEND SHOOTING_TIME
#> 1           1              S                  S          1114
#> 2           2              S                  S           948
#> 3           3              S                  S          1500
#> 4           4              S                  S          1355
#> 5           5              S                  S           502
#> 6           6              S                  S           615
#>   SHOOTING_QUADRANT SHOOTING_LATITUDE SHOOTING_LONGITUDE SHOOTING_DEPTH
#> 1                 1           4057.74            1340.01            376
#> 2                 1           4101.90            1341.50            109
#> 3                 1           4054.54            1349.77            113
#> 4                 1           4057.45            1347.27             95
#> 5                 1           4048.97            1357.72             77
#> 6                 1           4055.04            1356.21             25
#>   HAULING_TIME HAULING_QUADRANT HAULING_LATITUDE HAULING_LONGITUDE
#> 1         1214                1          4055.24           1342.01
#> 2         1018                1          4100.67           1342.29
#> 3         1530                1          4053.28           1351.54
#> 4         1425                1          4056.43           1348.39
#> 5          532                1          4050.38           1357.47
#> 6          645                1          4054.19           1355.04
#>   HAULING_DEPTH HAUL_DURATION VALIDITY COURSE RECORDED_SPECIES DISTANCE
#> 1           331            60        V      R                3     5500
#> 2           112            30        V      R                3     2593
#> 3           119            30        V      R                3     2926
#> 4            97            30        V      R                3     2500
#> 5            75            30        V      R                3     2759
#> 6            23            30        V      R                3     2630
#>   VERTICAL_OPENING WING_OPENING GEOMETRICAL_PRECISION BRIDLES_LENGTH
#> 1               19          191                     E            150
#> 2               19          182                     E            100
#> 3               19          182                     E            100
#> 4               19          182                     E            100
#> 5               19          182                     E            100
#> 6               19          163                     E            100
#>   WARP_LENGTH WARP_DIAMETER HYDROLOGICAL_STATION OBSERVATIONS
#> 1        1300            14                    0            0
#> 2         550            14                    0            0
#> 3         550            14                    0            0
#> 4         550            14                    0            0
#> 5         450            14                    0            0
#> 6         300            14                    0            0
#>   BOTTOM_TEMPERATURE_BEGINNING BOTTOM_TEMPERATURE_END MEASURING_SYSTEM
#> 1                         14.0                   14.0               VA
#> 2                         14.3                   14.2               VA
#> 3                         14.3                   14.0               VA
#> 4                         14.6                   14.3               VA
#> 5                         14.9                   14.9               VA
#> 6                         18.1                   18.6               VA
#>   NUMBER_OF_THE_STRATUM BOTTOM_SALINITY_BEGINNING BOTTOM_SALINITY_END
#> 1                 13404                        NA                  NA
#> 2                 13403                        NA                  NA
#> 3                 13403                        NA                  NA
#> 4                 13402                        NA                  NA
#> 5                 13402                        NA                  NA
#> 6                 13401                        NA                  NA
#>   MEASURING_SYSTEM_SALINITY
#> 1                        NA
#> 2                        NA
#> 3                        NA
#> 4                        NA
#> 5                        NA
#> 6                        NA
```

Two functions were also introduced to estimate the hauls distance using
respectivelly MEDITS coordinates and decimal degrees coordinates:

``` r
library(MEDITS)
# distance estimation using MEDITS coordinates
med.dist <- MEDITS.distance(TA, unit="km", verbose=TRUE)
#> kilometers
head(med.dist)
#> [1] 5.409739 2.531317 3.403627 2.454159 2.634717 2.271499

# distance estimation using decimal degrees coordinates
dd <- MEDITS.to.dd(TA)
dd.dist <- dd.distance(dd, unit="km", verbose=TRUE)
#> kilometers
head(dd.dist)
#> [1] 5.409739 2.531317 3.403627 2.454159 2.634717 2.271499
```

The function *land.points* allows to check whether some of the TA
table’s coordinates fall on the land:

``` r
land.points(TA)
#> 
#> None of the coordinates falls on the land
#> [1] NA
```

A plot of haul’s positions could also be generated, with the possibility
to define the years range to be considered, the GSA area and, whether
both initial and end coordinates should be plotted or only the mean haul
coordinates:

``` r
haul.plot(TA, GSA = 10, year = c(2007), type = "b", labels = TRUE)
```

<img src="man/figures/README-unnamed-chunk-5-1.png" width="100%" />

## Working with TB and TC files

To link the geographic information of the hauls respectively to catch
and biological data were introduced two different functions: *m.TATB*
and *m.TATC*:

``` r
# merge of TA and TB tables
mTATB <- m.TATB(TA,TB,"ARISFOL")

# merge of TA and TB tables
mTATC <- m.TATC(TA,TC,"ARISFOL")
```

The estimation of the indices’ time series is performed by the
*index.ts* function. It is possible to estimate the following indices:
abundance, inverse of abundance CV, biomass and mean individual weight
(MIW). The analysis could also be performed by sex.

``` r
merge_TATB <- m.TATB(TA,TB,"ARISFOL")
ind <- index.ts(merge_TATB, index="abundance", depth_range = c(500,800), sex="f",
str.scheme=strata_scheme, surf=stratification_scheme)
```

<img src="man/figures/README-unnamed-chunk-7-1.png" width="100%" />

The function *sexratio.ts* estimates the time series of the sex ratio:

``` r
ind <- m.TATB(TA,TB,"ARISFOL")
sexratio.ts(merge = ind, GSA = NA, sspp = NA, depth_range=c(500,800),
strata_scheme=strata_scheme, stratification=stratification_scheme, plot=TRUE )
```

<img src="man/figures/README-unnamed-chunk-8-1.png" width="100%" />

    #>    year Indices_F Indices_FM sex.ratio     variance         sd
    #> 1  2007  299.2558   486.8133 0.6147239 0.0009996861 0.03161781
    #> 2  2008  752.4499  1408.3255 0.5342869 0.0003541958 0.01882009
    #> 3  2009  767.4742  1481.7511 0.5179508 0.0003372210 0.01836358
    #> 4  2010 1299.4429  2274.6234 0.5712783 0.0002175715 0.01475031
    #> 5  2011  457.9915   911.2554 0.5025940 0.0005486862 0.02342405
    #> 6  2012  338.1038   667.3721 0.5066196 0.0007491415 0.02737045
    #> 7  2013  446.7506   921.2462 0.4849416 0.0005424969 0.02329156
    #> 8  2014  193.2033   339.2838 0.5694444 0.0014594096 0.03820222
    #> 9  2015  222.5422   375.6135 0.5924765 0.0013081896 0.03616890
    #> 10 2016  270.0096   481.6231 0.5606242 0.0010304969 0.03210135
    #>    positive_hauls_perc
    #> 1                  100
    #> 2                  100
    #> 3                  100
    #> 4                  100
    #> 5                  100
    #> 6                  100
    #> 7                  100
    #> 8                  100
    #> 9                  100
    #> 10                 100

The function *index.rs* allows to estimate the time series of abundance
index for both recruits and spawners:

``` r
index.rs (TA,TB,TC,GSA=10,sspp="ARISFOL",stage ="recruits",cutoff=29,
  depth_range=c(500,800),str.scheme=strata_scheme,
  surf=stratification_scheme,plot=TRUE)
```

<img src="man/figures/README-unnamed-chunk-9-1.png" width="100%" />

    #>    year   abundance         sd
    #> 1  2007  173.819218  85.441493
    #> 2  2008  746.525103 253.963668
    #> 3  2009  682.000011 295.301758
    #> 4  2010 1399.952318 724.374171
    #> 5  2011   73.869603  44.097009
    #> 6  2012  193.791201 106.736291
    #> 7  2013  194.214778 125.944553
    #> 8  2014    7.068412   3.177105
    #> 9  2015   75.946935  28.093725
    #> 10 2016   77.475985  38.404432

## Spatial analysis

Placing the above described indices in the spatial dimension is very
useful for several objectives, as the knowledge of the distribution and
abundance of the species at local and regional spatial scale, the
identification of areas with certain peculiarities, the localization of
sensitive life stage of the population, etc… GFCM grid is considered as
the better available trade-off for the spatial scale to be used in the
different steps of the analysis, especially considering the subsequent
possible overlap with maps of the fishing effort spatial distribution.
This despite the fact that not any unit in the grid can actually
encompass the spatial distribution of all the life stages of a given
species, especially when this distribution is depth related. To perform
the spatial analysis of the indicators it is needed to assign the haul
information to the relative cells of the GFCM statistical grid contained

NEWS.md
MEDITS v0.1.7
==============
- the print() function is used (F.I. in the case of LFD.R) to return the ggplot2 plots and not to return messages to the console. In cases in which messages() and warning() were used, they can be silenced with the paremeter verbose = TRUE.
- "T" and "F" substituted in all the scripts with "TRUE" and "FALSE" 
- The use of the *immediate* call of on.exit() was corrected in: sex.ratio.ts.R, index.rs.R, index.ts.R, sp.index.R, bubbleplot.R
- The acronyms (such as TA) were always explained in .Rd files
- In bubbleplot() the parameter "inches" was introduced to define the maximum radius (in inches) of the circles.


MEDITS v0.1.6
==============
-Saving option of plot eliminated from bubble.rs function
-Used "donttest" instead of "dontrun" in the following .Rd files: bubble.rs.Rd, TATB.grid, TATC.grid,haul.plot, index.rs, LFD, m.TATC, quant, sp.index.
-\value added to following .Rd files: bubble.rs.Rd, m.TATB.Rd, dd.distance.Rd, dd.to.MEDITS.Rd, MEDITS.distance.Rd, MEDITS.to.dd.Rd
-substitution of cat or print with message function in the following function: m.TATB, m.TATC, check.format, dd.distance, MEDITS.distance, land.points, index.ts
-verbose parameter added to the following functions: m.TATB, m.TATC, index.ts, dd.distance, MEDITS.distance
-added the on.exit() function to retain the user default values of par()  in the following functions: bubbleplot, index.rs, index.ts, sexratio.ts, sp.index
-in haul.plot function was introduced the possibility to select the format of the coordinate to be plotted: "MEDITS" or "degrees".
-estimation of abundance indices time series by sex in index.ts() function.
-q95() function substituted by quant() function that allows the estimation of the length at a user-defined percentile value of a LFD. The argument "quantile" is now required by the function.
-the outputs of LFD() function was modified. The LFD is returned as a data frame and not as the element of a list, as well as for the LFD plot.
-update of README.md file

MEDITS v0.1.5
==============
-The Title field formatted in title case

MEDITS v0.1.4
==============
- crop of wmap, countries, depth_1, depth_2, depth_3 shapefiles to the Mediterranean and Black Sea area to reduce package size

MEDITS v0.1.3
==============
- Title changed with "Analysis of MEDITS-like Survey Data" in the Description file 
- Link to MEDITS URL added in the the Description file

MEDITS v0.1.2
==============
- the function strata.area was suppressed because not stable 


R/index.rs.R


index.rs <- function (TA,TB,TC, GSA = NA, sspp , stage ="recruits", cutoff,  depth_range,  str.scheme=strata_scheme, surf=stratification, country=NA, plot=TRUE ) {   #,  strata_scheme, stratification
  if (FALSE){
    # require(ggplot2)
    GSA = NA
    sspp="ARISFOL"
    stage ="recruits"
    cutoff=29
    depth_range=c(200,800)
    str.scheme=strata_scheme
    surf=stratification_scheme
    country=NA
    plot=TRUE
    }
  strata_scheme <- str.scheme
  stratification <- surf
  if (is.na(sspp)) {
    GENERE <- as.character(unique(TB$GENUS)[unique(TB$GENUS) != -1])
    SPECIE <- as.character(unique(TB$SPECIES)[unique(TB$SPECIES) != -1])
    sspp <- paste(GENERE,SPECIE, sep="")
    species <- sspp
  } else {
    species <- sspp
  }


  if (is.na(GSA)) {
    GSA <- unique(TA$AREA)
  }


  merge_TATB <- m.TATB(TA,TB,sspp)
  merge_TATC <- m.TATC(TA,TC,sspp)

  merge_TATB <- merge_TATB[merge_TATB$GSA == GSA , ]
  merge_TATC <- merge_TATC[merge_TATC$GSA == GSA , ]

  #-----------------------
  # ABUNDANCE INDICES
  #-----------------------

  dependent <- "abundance"


dep_text <-expression(paste(Abundance, (n/km^2), sep=" "))
varcol <- which(colnames(merge_TATC)=="N_km2")
col_response <- which(colnames(merge_TATC)=="N_km2")


threshold <- cutoff

sex <- "F"

#------------------------------------------
# selection of depth range

if (is.na(country)){
  data <- merge_TATC[merge_TATC$GSA == GSA , ]
  strata.scheme <- strata_scheme[strata_scheme$GSA == GSA & strata_scheme$COUNTRY == as.character(unique(merge_TATB$COUNTRY)[1]), ]
}else{
  if (!is.na(country)){
    data <- merge_TATC[merge_TATC$GSA == GSA &  merge_TATC$COUNTRY == country, ]
    strata.scheme <- strata_scheme[strata_scheme$GSA == GSA & strata_scheme$COUNTRY == country, ]
  }
}

# strata <- unique(strata.scheme$CODE)
# s <- data.frame(matrix(NA,ncol=5,nrow=1))
# colnames(s) <- colnames(strata.scheme)
# i=1
# for (i in 1:length(strata)){
#   s[i,] <- strata.scheme[(strata.scheme$CODE == strata[i]),][1,]
#   s[i,2] <- as.character(strata.scheme[(strata.scheme$CODE == strata[i]),2][1])
# }
#
# strata.scheme <- s

depth <- data.frame(matrix(NA,ncol=4, nrow=length(strata.scheme$CODE)))
colnames(depth) <- c("strata", "min", "max", "bul")
depth$strata <- strata.scheme$CODE # c(1,2,3,4,5)
depth$min <-strata.scheme$MIN_DEPTH  # c(10,50,100,200,500)
depth$max <-strata.scheme$MAX_DEPTH  # c(50,100,200,500,800)


depth_range <- data.frame(depth_range); depth_range <- as.numeric(as.character(depth_range[,1]))
if (depth_range[2] != 800) {depth_range[2] <- depth_range[2]}
data <-  data[data$MEAN_DEPTH>depth_range[1] & data$MEAN_DEPTH<=depth_range[2],]
strata_range <- seq(which(depth[,"min"]==depth_range[1]),which(depth[,"max"]==depth_range[2]),1)
depth[depth$strata %in% strata_range, "bul"] <- TRUE
depth[!(depth$strata %in% strata_range), "bul"] <- FALSE

analysis_stratum1 <- NA
analysis_stratum2 <- NA
analysis_stratum3 <- NA
analysis_stratum4 <- NA
analysis_stratum5 <- NA
analysis_stratum5 <- NA

analysis_stratum1 <- depth[1,4]
analysis_stratum2 <- depth[2,4]
analysis_stratum3 <- depth[3,4]
analysis_stratum4 <- depth[4,4]
analysis_stratum5 <- depth[5,4]
analysis_stratum6 <- depth[6,4]

if (is.na(analysis_stratum1)){depth[1,4]<- FALSE; analysis_stratum1 <- FALSE}
if (is.na(analysis_stratum2)){depth[2,4]<- FALSE; analysis_stratum2 <- FALSE}
if (is.na(analysis_stratum3)){depth[3,4]<- FALSE; analysis_stratum3 <- FALSE}
if (is.na(analysis_stratum4)){depth[4,4]<- FALSE; analysis_stratum4 <- FALSE}
if (is.na(analysis_stratum5)){depth[5,4]<- FALSE; analysis_stratum5 <- FALSE}
if (is.na(analysis_stratum6)){depth[6,4]<- FALSE; analysis_stratum6 <- FALSE}

#------------------------------------------


data$n_raised <- data$NUMBER_OF_INDIVIDUALS_IN_THE_LENGTH_CLASS_AND_MATURITY_STAGE * data$RAISING_FACTOR

if (stage=="recruits"){
ddd <- data[data$LENGTH_CLASS <= threshold,]
} else { if (stage=="spawners"){
  ddd <- data[data$LENGTH_CLASS >= threshold  & data$SEX ==sex  ,]
}
}


ddd_n <- aggregate(ddd$n_raised, by=list(ddd$id), sum) # [,1])
colnames(ddd_n) <- c("id","sum_N")
ddd_nkm2 <-  aggregate(ddd$N_km2, by=list(ddd$id), sum)
colnames(ddd_nkm2) <- c("id","sum_nkm2")
merge_N <- merge(ddd_n, ddd_nkm2, by.x="id")


ddd <- merge(merge_TATB[,c("id","GSA","YEAR","HAUL_NUMBER","GENUS","SPECIES","MEAN_DEPTH","SWEPT_AREA")], merge_N, by.y="id",all.x =TRUE)   # 1,3,4,6,22,23,34,35
ddd <- ddd[ddd$MEAN_DEPTH >=depth_range[1] & ddd$MEAN_DEPTH < depth_range[2],]

for (i in 1:length(ddd$sum_N)){
  if (is.na(ddd[i,"sum_N" ])) {ddd[i,"sum_N" ] <- 0} else {ddd[i,"sum_N" ] <- ddd[i,"sum_N" ]}
  if (is.na(ddd[i,"sum_nkm2" ])) {ddd[i,"sum_nkm2" ] <- 0} else {ddd[i,"sum_nkm2" ] <- ddd[i,"sum_nkm2" ]}
}

year_range <- data.frame(year = sort(unique(merge_TATC$YEAR)))

res_table_cala2 <- data.frame(year=sort(unique(merge_TATC$YEAR)))
res_table_cala2$index_1 <- NA
res_table_cala2$index_2 <- NA
res_table_cala2$index_3 <- NA
res_table_cala2$index_4 <- NA
res_table_cala2$index_5 <- NA
res_table_cala2$index_6 <- NA

se_table2  <-  data.frame(year=sort(unique(merge_TATC$YEAR)))
se_table2$sum_1 <- NA
se_table2$sum_2 <- NA
se_table2$sum_3 <- NA
se_table2$sum_4 <- NA
se_table2$sum_5 <- NA
se_table2$sum_6 <- NA
se_table2$sd <- NA

se_table_a  <-  data.frame(year=sort(unique(merge_TATC$YEAR)))
se_table_a$sum_1 <- NA
se_table_a$sum_2 <- NA
se_table_a$sum_3 <- NA
se_table_a$sum_4 <- NA
se_table_a$sum_5 <- NA
se_table_a$sum_6 <- NA
se_table_a$sd <- NA

if (is.na(country)){
  if (analysis_stratum1 == TRUE){area_s1 <- sum(stratification[stratification$GSA == GSA & stratification$CODE==1,5])} else {area_s1 <- 0}
  if (analysis_stratum2 == TRUE){area_s2 <- sum(stratification[stratification$GSA == GSA & stratification$CODE==2,5])} else {area_s2 <- 0}
  if (analysis_stratum3 == TRUE){area_s3 <- sum(stratification[stratification$GSA == GSA & stratification$CODE==3,5])} else {area_s3 <- 0}
  if (analysis_stratum4 == TRUE){area_s4 <- sum(stratification[stratification$GSA == GSA & stratification$CODE==4,5])} else {area_s4 <- 0}
  if (analysis_stratum5 == TRUE){area_s5 <- sum(stratification[stratification$GSA == GSA & stratification$CODE==5,5])} else {area_s5 <- 0}
  if (analysis_stratum6 == TRUE){area_s6 <- sum(stratification[stratification$GSA == GSA & stratification$CODE==6,5])} else {area_s6 <- 0}
} else {
  if (analysis_stratum1 == TRUE){area_s1 <- sum(stratification[stratification$GSA == GSA & stratification$COUNTRY == country & stratification$CODE==1,5])} else {area_s1 <- 0}
  if (analysis_stratum2 == TRUE){area_s2 <- sum(stratification[stratification$GSA == GSA & stratification$COUNTRY == country & stratification$CODE==2,5])} else {area_s2 <- 0}
  if (analysis_stratum3 == TRUE){area_s3 <- sum(stratification[stratification$GSA == GSA & stratification$COUNTRY == country & stratification$CODE==3,5])} else {area_s3 <- 0}
  if (analysis_stratum4 == TRUE){area_s4 <- sum(stratification[stratification$GSA == GSA & stratification$COUNTRY == country & stratification$CODE==4,5])} else {area_s4 <- 0}
  if (analysis_stratum5 == TRUE){area_s5 <- sum(stratification[stratification$GSA == GSA & stratification$COUNTRY == country & stratification$CODE==5,5])} else {area_s5 <- 0}
  if (analysis_stratum6 == TRUE){area_s6 <- sum(stratification[stratification$GSA == GSA & stratification$COUNTRY == country & stratification$CODE==6,5])} else {area_s6 <- 0}
}

peso_s1 <- area_s1/sum(area_s1,area_s2,area_s3,area_s4,area_s5,area_s6)
peso_s2 <- area_s2/sum(area_s1,area_s2,area_s3,area_s4,area_s5,area_s6)
peso_s3 <- area_s3/sum(area_s1,area_s2,area_s3,area_s4,area_s5,area_s6)
peso_s4 <- area_s4/sum(area_s1,area_s2,area_s3,area_s4,area_s5,area_s6)
peso_s5 <- area_s5/sum(area_s1,area_s2,area_s3,area_s4,area_s5,area_s6)
peso_s6 <- area_s6/sum(area_s1,area_s2,area_s3,area_s4,area_s5,area_s6)

i=1
for(i in 1:length(res_table_cala2[,1])) {


  data2 <- ddd[ddd$YEAR == res_table_cala2[i,1] , ]

  if (analysis_stratum1 == TRUE) {s1 <- data2[data2$MEAN_DEPTH >= depth[1,2] & data2$MEAN_DEPTH < depth[1,3], ]}
  if (analysis_stratum2 == TRUE) {s2 <- data2[data2$MEAN_DEPTH >= depth[2,2] & data2$MEAN_DEPTH < depth[2,3], ]}
  if (analysis_stratum3 == TRUE) {s3 <- data2[data2$MEAN_DEPTH >= depth[3,2] & data2$MEAN_DEPTH < depth[3,3], ]}
  if (analysis_stratum4 == TRUE) {s4 <- data2[data2$MEAN_DEPTH >= depth[4,2] & data2$MEAN_DEPTH < depth[4,3], ]}
  if (analysis_stratum5 == TRUE) {s5 <- data2[data2$MEAN_DEPTH >= depth[5,2] & data2$MEAN_DEPTH < depth[5,3], ]}
  if (analysis_stratum6 == TRUE) {s6 <- data2[data2$MEAN_DEPTH >= depth[6,2] & data2$MEAN_DEPTH < depth[6,3], ]}

  if (analysis_stratum1 == TRUE){
    s_n <- sum(s1[!is.na(s1$sum_N),"sum_N"])
    s_a <- sum(s1$SWEPT_AREA)
    res_table_cala2[i,2] <- s_n/s_a

    mean_ind <- mean(s1[!is.na(s1$sum_nkm2), "sum_nkm2" ])
    sq <- (s1[!is.na(s1$sum_nkm2),"sum_nkm2"] - mean_ind)^2
    sqA <- sq*s1$SWEPT_AREA
    sum_sqA <- sum(sqA)/(length(s1[,1])-1)
    f1 <- s_a/area_s1
    se_table2[i,2] <- (((peso_s1)^2 * sum_sqA)/ s_a)* (1-f1)

  } else {area_s1 = 0}

  if (analysis_stratum2 == TRUE){
    s_n <- sum(s2[!is.na(s2$sum_N),"sum_N"])
    s_a <- sum(s2$SWEPT_AREA)
    res_table_cala2[i,3] <- s_n/s_a

    mean_ind <- mean(s2[!is.na(s2$sum_nkm2), "sum_nkm2" ])
    sq <- (s2[!is.na(s2$sum_nkm2),"sum_nkm2"] - mean_ind)^2
    sqA <- sq*s2$SWEPT_AREA
    sum_sqA <- sum(sqA)/(length(s2[,1])-1)
    f2 <- s_a/area_s2
    se_table2[i,3] <- (((peso_s2)^2 * sum_sqA)/ s_a)* (1-f2)
  } else {area_s2 = 0}

  if (analysis_stratum3 == TRUE){
    s_n <- sum(s3[!is.na(s3$sum_N),"sum_N"])
    s_a <- sum(s3$SWEPT_AREA)
    res_table_cala2[i,4] <- s_n/s_a

    mean_ind <- mean(s3[!is.na(s3$sum_nkm2), "sum_nkm2" ])
    sq <- (s3[!is.na(s3$sum_nkm2),"sum_nkm2"] - mean_ind)^2
    sqA <- sq*s3$SWEPT_AREA
    sum_sqA <- sum(sqA)/(length(s3[,1])-1)
    f3 <- s_a/area_s3
    se_table2[i,4] <- (((peso_s3)^2 * sum_sqA)/ s_a)* (1-f3)
  } else {area_s3 = 0}

  if (analysis_stratum4 == TRUE){
    s_n <- sum(s4[!is.na(s4$sum_N),"sum_N"])
    s_a <- sum(s4$SWEPT_AREA)
    res_table_cala2[i,5] <- s_n/s_a

    mean_ind <- mean(s4[!is.na(s4$sum_nkm2), "sum_nkm2" ])
    sq <- (s4[!is.na(s4$sum_nkm2),"sum_nkm2"] - mean_ind)^2
    sqA <- sq*s4$SWEPT_AREA
    sum_sqA <- sum(sqA)/(length(s4[,1])-1)
    f4 <- s_a/area_s4
    se_table2[i,5] <- (((peso_s4)^2 * sum_sqA)/ s_a)* (1-f4)
  } else {area_s4 = 0}

  if (analysis_stratum5 == TRUE){
    #index computation
    s_n <- sum(s5[!is.na(s5$sum_N),"sum_N"])
    s_a <- sum(s5$SWEPT_AREA)
    res_table_cala2[i,6] <- s_n/s_a

    #se computation
    mean_ind <- mean(s5[!is.na(s5$sum_nkm2), "sum_nkm2" ])
    sq <- (s5[!is.na(s5$sum_nkm2),"sum_nkm2"] - mean_ind)^2
    sqA <- sq*s5$SWEPT_AREA
    sum_sqA <- sum(sqA)/(length(s5[,1])-1)
    f5 <- s_a/area_s5
    se_table2[i,6] <- (((peso_s5)^2 * sum_sqA)/ s_a)* (1-f5)
  } else {area_s5 = 0}

  if (analysis_stratum6 == TRUE){
    #index computation
    s_n <- sum(s6[!is.na(s6$sum_N),"sum_N"])
    s_a <- sum(s6$SWEPT_AREA)
    res_table_cala2[i,7] <- s_n/s_a

    #se computation
    mean_ind <- mean(s6[!is.na(s6$sum_nkm2), "sum_nkm2" ])
    sq <- (s6[!is.na(s6$sum_nkm2),"sum_nkm2"] - mean_ind)^2
    sqA <- sq*s6$SWEPT_AREA
    sum_sqA <- sum(sqA)/(length(s6[,1])-1)
    f6 <- s_a/area_s6
    se_table2[i,7] <- (((peso_s6)^2 * sum_sqA)/ s_a)* (1-f6)
  } else {area_s6 = 0}

  # source(paste(wd, "/scripts/Index_estimation_cala_recruits.R", sep=""), encoding = 'UTF-8')
  sum_res_est <- c(res_table_cala2[i,2]*peso_s1,res_table_cala2[i,3]*peso_s2,res_table_cala2[i,4]*peso_s3,res_table_cala2[i,5]*peso_s4,res_table_cala2[i,6]*peso_s5,res_table_cala2[i,7]*peso_s6)
  res_table_cala2[i, 8]<- sum(sum_res_est[!is.na(sum_res_est)])# /sum(area_s1,area_s2,area_s3,area_s4,area_s5)
  colnames(res_table_cala2) <- c("year", "stratum 1","stratum 2", "stratum 3", "stratum 4", "stratum 5", "stratum 6", "Indices")
  se_table2[i, "sd"] <-sqrt(rowSums(se_table2[i, 2:7], na.rm = TRUE))
}

timeseries <- data.frame(year = res_table_cala2[,1], abundance = res_table_cala2[,8], sd= se_table2[, "sd"])

if (plot){
  oldoptions <- options()$warn
  old_par <- list()
  old_par$mfrow <-par()$mfrow
  old_par$oma <-par()$oma
  old_par$mgp <-par()$mgp

# plot timeseries of mean spawners indices
main <- paste(sspp,"_GSA",GSA,"_(abundance of ", stage,")_",depth_range[1],"-",depth_range[2], " m", sep="")
main.lab <- paste(sspp," GSA",GSA," (abundance of ", stage,")_",depth_range[1],"-",depth_range[2], " m", sep="")
max_index <- max(timeseries[,2]) + max(timeseries[!is.na(timeseries$sd),3]*1.5)

on.exit(c(par(mfrow=old_par$mfrow, oma=old_par$oma, mgp=old_par$mgp), options(warn=oldoptions)))
  options(warn=-1)
par(mfrow=c(1,1),oma=c(1,1,1,1), mgp=c(2, 1,0))
plot(timeseries[,1],  timeseries[,2], type="b", col="black", pch=16, xlab="year", ylim=c(0,max_index*1.2), ylab=dep_text, main=main.lab) # ylim=c(0,max_index*1.2)
lines(timeseries[,1], (timeseries[,2]-1.96*timeseries[,3]), type="l",lty=2, col="red" )
lines(timeseries[,1], (timeseries[,2]+1.96*timeseries[,3]), type="l",lty=2, col="red" )
legend("topright", c("time series", "CI"), lty=c(1,2), pch=c(16, NA), col=c("black","red"))

}

return(timeseries)
}
